src/main/java/com/mraof/minestuck/item/weapon/GivePotionEffectWeaponItem.java

```diff
@@ -0,0 +1,33 @@
+package com.mraof.minestuck.item.weapon;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.IItemTier;
+import net.minecraft.item.ItemStack;
+import net.minecraft.potion.Effect;
+import net.minecraft.potion.EffectInstance;
+
+public class GivePotionEffectWeaponItem extends WeaponItem
+{
+    private Effect effect;
```

Comment:
Oh, and Effects are registry object that are initialized after items. While it'll be fine for vanilla effects, if we want to support mod effects these should be changed to effect suppliers. It's not a big deal right now, but do it if you think it's good to have it done already.